src/Data/List/NonEmpty/Base.agda

```diff
@@ -203,9 +203,12 @@ snocView (x ∷ .(xs List.∷ʳ y)) | xs List.∷ʳ′ y = (x ∷ xs) ∷ʳ′ y
 
 -- The last element in the list.
 
+private
+  last′ : ∀ {l} → SnocView {A = A} l → A
+  last′ (_ ∷ʳ′ y) = y
+
 last : List⁺ A → A
-last xs with snocView xs
-last .(ys ∷ʳ y) | ys ∷ʳ′ y = y
+last = last′ ∘ snocView
```

Comment:
I'd have been happy with irrefutable `with` (the us eof the dot pattern in the original is for 'commentary' purposes only, I think), but I take your (earlier) comment about (projections from) one-constructor datatypes